org.quantil.qprov.collector/src/main/java/org/quantil/qprov/collector/providers/aws/AWSProvider.java

```diff
@@ -0,0 +1,747 @@
+            logger.debug("Stored characteristics are up-to-date. No update needed!");
+            return;
+        }
+
+        // create new characteristics object with the current characteristics
+        final QubitCharacteristics qubitCharacteristics = new QubitCharacteristics();
+        qubitCharacteristics.setQubit(currentQubit);
+        qubitCharacteristics.setCalibrationTime(calibrationTime);
+
+        // retrieve T1, T2, and readout error
+        switch (device.getProviderName().toLowerCase()) {
+            case "ionq":
+                handleIonqQubitProperties(qubitCharacteristics, device);
+                break;
+            case "rigetti":
+                handleRigettiQubitProperties(qubitCharacteristics, device);
+                break;
+            default:
+                logger.warn("For device {} of provider {} no qubit handler is available. Qubit properties will be null.", device.getDeviceName(), device.getProviderName());
+        }
+
+        // update qubit object with new characteristics object
+        currentQubit.getQubitCharacteristics().add(qubitCharacteristics);
+        qubitRepository.save(currentQubit);
+    }
+
+    private void handleRigettiQubitProperties(QubitCharacteristics qubitCharacteristics, AWSDevice device) {
+        logger.warn("Retrieving Qubit properties for Rigetti devices not yet supported!");
+        // TODO: Implement me
+    }
+
+    private void handleIonqQubitProperties(QubitCharacteristics qubitCharacteristics, AWSDevice device) {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            JsonNode providerNode = mapper.readTree(device.getDeviceCapabilities()).get("provider");
+            if (providerNode == null) {
+                logger.warn("The IONQ device json has not the expected format. Cannot find provider node.");
+                return;
+            }
+            // Fidelities
+            JsonNode fidelityNode = providerNode.get("fidelity");
+            if (fidelityNode != null) {
+                JsonNode readoutError = fidelityNode.get("spam");
+                if (readoutError == null) {
+                    logger.warn("The IONQ device json has not the expected format. Cannot find spam/readout error node.");
+                } else {
+                    qubitCharacteristics.setReadoutError(BigDecimal.valueOf(1.0).subtract(new BigDecimal(readoutError.get("mean").asText())));
+                }
+            } else {
+                logger.warn("The IONQ device json has not the expected format. Cannot find fidelity node.");
+            }
+            JsonNode timingNode = providerNode.get("timing");
+            if (timingNode != null) {
+                JsonNode t1 = timingNode.get("T1");
+                JsonNode t2 = timingNode.get("T2");
+                if (t1 != null) {
+                    // in seconds
+                    qubitCharacteristics.setT1Time(BigDecimal.valueOf(t1.asDouble()));
```

Comment:
@wederbn Here we got the following issue: AWS returns all timing information in seconds instead of microseconds. But if I try to manipulate the scale to microseconds I run into a numeric overflow with the schema:
`A field with precision 20, scale 15 must round to an absolute value less than 10^5`
IThe issue seems to be that the default decimal scale is 15 which only leaves 5 decimal places in front of the decimal point. How should we handle this?

Reply:
Marie said it should be long double as data type

Reply:
EDIT: double would round the times and error rates - we want to be precise, thus, the assumption is to adapt the BigDecimal constants 
https://github.com/UST-QuAntiL/qprov/blob/22a303d04cbe5768550187e2441cb9272c27634e/org.quantil.qprov.core/src/main/java/org/quantil/qprov/core/Constants.java#L30